nifi-nar-bundles/nifi-airtable-bundle/nifi-airtable-processors/src/test/java/org/apache/nifi/processors/airtable/QueryAirtableTableIT.java

```diff
@@ -0,0 +1,177 @@
+        runner.addControllerService("webClientService", webClientServiceProvider);
+        runner.enableControllerService(webClientServiceProvider);
+
+        runner.setProperty(QueryAirtableTable.API_KEY, "???");
+        runner.setProperty(QueryAirtableTable.BASE_ID, "appabcdefghijklmn");
+        runner.setProperty(QueryAirtableTable.TABLE_ID, "tblabcdefghijklmn");
+        runner.setProperty(QueryAirtableTable.SCHEMA_READER, schemaReader.getIdentifier());
+        runner.setProperty(QueryAirtableTable.RECORD_WRITER, writer.getIdentifier());
+        runner.setProperty(QueryAirtableTable.WEB_CLIENT_SERVICE_PROVIDER, webClientServiceProvider.getIdentifier());
+
+    }
+
+    @AfterEach
+    void tearDown() {
+        runner.shutdown();
+    }
+
+    @Test
+    void retrievesAndWritesRecords() throws Exception {
+        try (final MockWebServer server = new MockWebServer()) {
+            server.enqueue(new MockResponse().setBody(RECORDS_JSON_BODY));
+
+            server.start();
+            final HttpUrl httpUrl = server.url("/v0/airtable");
+
+            runner.setProperty(QueryAirtableTable.API_URL, httpUrl.toString());
+            runner.run();
+
+            final List<MockFlowFile> results = runner.getFlowFilesForRelationship(QueryAirtableTable.REL_SUCCESS);
+            assertEquals(1, results.size());
+            final MockFlowFile flowFile = results.get(0);
+            assertEquals("1", flowFile.getAttribute("record.count"));
+            final String content = flowFile.getContent();
+            assertEquals(EXPECTED_RECORD_CONTENT, content);
+        }
+    }
+
+    @Test
+    void retrievesAndWritesPagedRecords() throws Exception {
+        try (final MockWebServer server = new MockWebServer()) {
+            server.enqueue(new MockResponse().setBody(RECORDS_WITH_OFFSET_JSON_BODY));
+            server.enqueue(new MockResponse().setBody(RECORDS_JSON_BODY));
+
+            server.start();
+            final HttpUrl httpUrl = server.url("/v0/airtable");
+
+            runner.setProperty(QueryAirtableTable.API_URL, httpUrl.toString());
+            runner.run();
+
+            final List<MockFlowFile> results = runner.getFlowFilesForRelationship(QueryAirtableTable.REL_SUCCESS);
+            assertEquals(1, results.size());
+            final MockFlowFile flowFile = results.get(0);
+            assertEquals("2", flowFile.getAttribute("record.count"));
+            final String content = flowFile.getContent();
+            assertEquals(EXPECTED_RECORD_CONTENT + EXPECTED_RECORD_CONTENT, content);
+        }
+    }
+
+    @Test
+    void retrievesAndWritesPagedRecordsInMultipleFlowFiles() throws Exception {
+        try (final MockWebServer server = new MockWebServer()) {
+            server.enqueue(new MockResponse().setBody(RECORDS_WITH_OFFSET_JSON_BODY));
+            server.enqueue(new MockResponse().setBody(RECORDS_JSON_BODY));
+
+            server.start();
+            final HttpUrl httpUrl = server.url("/v0/airtable");
```

Comment:
Recommend declaring `/v0/airtable` as a static member variable and reusing across all test methods.